modules/imgproc/src/intersection.cpp

```diff
@@ -114,10 +122,20 @@ static int _rotatedRectangleIntersection( const RotatedRect& rect1, const Rotate
             float vx2 = vec2[j].x;
             float vy2 = vec2[j].y;
 
+            float numericalScalingFactor = std::min(std::abs(vx1*vx1+vy1*vy1), std::abs(vx2*vx2+vy2*vy2));
+            numericalScalingFactor = !numericalScalingFactor ? 1.f : 1.f/(numericalScalingFactor);
+            if (std::isinf(numericalScalingFactor))
+                numericalScalingFactor = 1.f;
+
+            vx1 *= numericalScalingFactor;
+            vy1 *= numericalScalingFactor;
+            vx2 *= numericalScalingFactor;
+            vy2 *= numericalScalingFactor;
+
             float det = vx2*vy1 - vx1*vy2;
 
-            float t1 = (vx2*y21 - vy2*x21) / det;
-            float t2 = (vx1*y21 - vy1*x21) / det;
+            float t1 = (vx2*y21 - vy2*x21)*numericalScalingFactor/det;
+            float t2 = (vx1*y21 - vy1*x21)*numericalScalingFactor/det;
```

Comment:
We already have normalization multiplier in the first section. Why do we apply it twice for t1/t2?

Reply:
Agreed.

However more clear code should multiply:

```
x21 *= numericalScalingFactor;
y21 *= numericalScalingFactor;
```

This code has 2 multiply operations (but could be executed in parallel).

---

Alternative is to replace 2 mul + 2 div ops => 2 mul + 1 div

```
float det_inv = numericalScalingFactor/det;

float t1 = (...)*det_inv;
float t2 = (...)*det_inv;
```

Bailout check could be changed too:
```.diff
-if( cvIsInf(t1) || cvIsInf(t2) || cvIsNaN(t1) || cvIsNaN(t2) )
+if (cvIsInf(dev_inv))
     continue;
```

But need to analyze numerical stability of this code first.

---


BTW, this code section is a generic "Solve for 2x2 Ax=b". Perhaps we could extract its code as a separate function (somewhere into core module).